### PR TITLE
9 22 user create

### PR DESCRIPTION
**Pull request by kcockett**
### Description of changes
Adds new user page, User controller, User model, user creation functionality, navigation link updates, user Dashboard page [skeleton], feature tests with happy/sad path, and model test

Page formatting will be applied when merged to main

### Screenshots (if applicable) of relevant views
image

### Test Suite
- [X] Are all tests within the test suite passing?
- [X] 100% coverage

#### Add GIF (NOT Optional)
https://media.giphy.com/media/Pk9dX0ZrzVQQGC6jUG/giphy.gif

### CODE REVIEW

**Review by z-fitch — APPROVED** [review]
Looks good!

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  end
+  
+  private
+
+  def user_params
+    params.permit(:username, :password)
+  end
+  
+  def user_params_valid?
+    !user_params[:username].empty? && !user_params[:password].empty? && user_params[:password] == params[:password_verify]
```
Comment (westonio) [inline app/controllers/users_controller.rb:29]: Out of curiousity - would the model's validations not take care of both `!user_params[:username].empty? && !user_params[:password].empty?` ?

**Review by westonio — APPROVED** [review]
Everything looks good! I have just added one note for you @kcockett in regards to some logic that may or may not be necessary with model validations.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (kcockett) [reply app/controllers/users_controller.rb:29]: Yes, I suppose it would.  Then I would implement a rescue block to capture the error.

**Pull request merged**